src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/TableSaw.java

```diff
@@ -54,28 +63,44 @@ public TableSaw(Category category, SlimefunItemStack item) {
     }
 
     @Override
-    public void onInteract(Player p, Block b) {
-        ItemStack log = p.getInventory().getItemInMainHand();
-
-        Optional<Material> planks = MaterialConverter.getPlanksFromLog(log.getType());
+    public void onInteract(@Nonnull Player p, @Nonnull Block b) {
+        ItemStack item = p.getInventory().getItemInMainHand();
+        ItemStack output = getItemsToOutput(item.getType());
 
-        if (planks.isPresent()) {
-            if (p.getGameMode() != GameMode.CREATIVE) {
-                ItemUtils.consumeItem(log, true);
-            }
+        if (output == null) {
+            return;
```

Comment:
Just a general thought: Should we maybe send a message or play a sound effect here?

Reply:
A sound effect is qlready played in the end. Playing 2 sounds would just confuse players.

Reply:
Play a sound when the output is null you mean?

Reply:
Just an indicator that this item couldn't be used, either some sound that conveys "failure" XD, maybe smoke particles or just the usual "not a valid recipe" message.
It's just a general idea.

Reply:
Ill do the generic message